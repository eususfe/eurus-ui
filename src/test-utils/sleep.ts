/*
 * @Author: jiangxd
 * @Date: 2022-09-23 14:06:55
 * @LastEditTime: 2022-09-23 14:07:04
 * @LastEditors: jiangxd
 * @Description:
 * @FilePath: /eurus-ui/src/test-utils/sleep.ts
 */
const sleep = (time = 0) => new Promise(resolve => setTimeout(resolve, time));

export default sleep;
