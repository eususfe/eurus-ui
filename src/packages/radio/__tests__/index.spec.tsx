import { mount } from '@vue/test-utils';
import { expect } from 'vitest';
import { h } from 'vue';
import ERadio, { ERadioGroup } from '..';

describe('eRadio', () => {
	it('should emit change event', async () => {
		const fn = vi.fn();
		const wrapper = mount(ERadio, {
			props: {
				modelValue: '1',
				value: '1',
				label: '1',
				onChange: fn,
			},
		});

		await wrapper.find('input').trigger('change');
		expect(wrapper.html()).toMatchSnapshot();
		expect(fn).toHaveBeenCalled();
	});
	it('should work with disabled props', async () => {
		const fn = vi.fn();
		const wrapper = mount(ERadio, {
			props: {
				modelValue: '1',
				value: '1',
				label: '1',
				disabled: true,
				onChange: fn,
			},
		});
		const groupWrapper = mount(ERadioGroup, {
			props: {
				modelValue: '1',
				disabled: true,
				onChange: fn,
			},
			slots: {
				default: () => h(ERadio, { label: '1', value: '1' }),
			},
		});
		await wrapper.trigger('click');
		await groupWrapper.findComponent(ERadio).trigger('click');
		expect(wrapper.html()).toMatchSnapshot();
		expect(groupWrapper.html()).toMatchSnapshot();
		expect(fn).not.toHaveBeenCalled();
	});
});
