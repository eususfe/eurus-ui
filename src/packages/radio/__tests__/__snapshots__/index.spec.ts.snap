// Vitest Snapshot v1

exports[`ERadio > ERadio snapshot 1`] = `
"<label class=\\"eu-radio\\"><span><span class=\\"radio-inner\\"></span>
  <!---->
  <!----></span>
</label>"
`;
